usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -192,6 +195,50 @@ spa_history_zone(void)
 }
 
 /*
+ * Post a history sysevent.  The nvlist_t* passed into this function will be
+ * duplicated (by nvlist_add_nvlist) and should be freed by caller.
+ * This function will only run in the context of the kernel.
+ */
+static void
+spa_history_log_notify(spa_t *spa, nvlist_t *nvl)
```

Comment:
I agree with what @grwilson said - this code duplicates `spa_event_create()`.  We should instead modify `spa_event_create()` to add a `nvlist_t*` argument, and call that from here.